test/get-group-tangle.test.js

```diff
@@ -0,0 +1,240 @@
+              if (err) throw err
+              t.deepEqual(
+                { root, previous },
+                { root: rootKey, previous: [msg.key] },
+                'adding message to root'
+              )
+
+              server.tribes2.publish(content, (err, msg) => {
+                if (err) throw err
+
+                getGroupTangle(group.id, (err, { root, previous }) => {
+                  if (err) throw err
+                  t.deepEqual(
+                    { root, previous },
+                    { root: rootKey, previous: [msg.key] },
+                    'adding message to tip'
+                  )
+                  server.close(true, t.end)
+                })
+              })
+            })
+          })
+        })
+      )
+    })
+  })
+})
+
+const n = 100
+test(`get-group-tangle-${n}-publishes`, (t) => {
+  const publishArray = new Array(n).fill().map((item, i) => i)
+  const server = Testbot()
+  let count = 0
+  server.tribes2.create(null, (err, data) => {
+    if (err) throw err
+
+    const groupId = data.id
+    pull(
+      pull.values(publishArray),
+      paraMap(
+        (value, cb) =>
+          server.tribes2.publish({ type: 'memo', value, recps: [groupId] }, cb),
+        4
+      ),
+      paraMap((msg, cb) => server.db.getMsg(msg.key, cb), 10),
+      pull.drain(
+        (m) => {
+          count += m.value.content.tangles.group.previous.length
+        },
+        (err) => {
+          t.error(err, 'no error')
+
+          // t.equal(count, n, 'We expect there to be no branches in our groupTangle')
+          t.true(
+            count < n * 8,
+            'We expect bounded branching with fast publishing'
+          )
+
+          server.close(true, t.end)
+        }
+      )
+    )
+  })
+})
+
+test('get-group-tangle', (t) => {
+  const tests = [
+    {
+      plan: 4,
+      test: (t) => {
+        const DESCRIPTION = 'auto adds group tangle'
+        // this is an integration test, as get-group-tangle is used in ssb.tribes2.publish
+        const ssb = Testbot()
+
+        ssb.tribes2.create(null, (err, data) => {
+          t.error(err, 'create group')
+
+          const groupRoot = data.root
+          const groupId = data.id
+
+          const content = {
+            type: 'yep',
+            recps: [groupId],
+          }
+
+          ssb.tribes2.publish(content, (err, msg) => {
+            t.error(err, 'publish a message')
+
+            ssb.db.get(msg.key, (err, A) => {
+              t.error(err, 'get that message back')
+
+              //TODO: this is confusing, why is previous supposed to be groupRoot? because right after creating the group we add ourselves to the group, which creates another message, so that message should be the previous
+              //t.deepEqual(
+              //  A.content.tangles.group, // actual
+              //  { root: groupRoot, previous: [groupRoot] }, // expected
+              //  DESCRIPTION + ' (auto added tangles.group)'
+              //)
```

Comment:
Once I added this https://github.com/ssbc/ssb-tribes2/pull/4/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R73-R75 (adding self to group when you've just created the group) this test (from tribes1) started failing. The admin is also self-added in tribes1 so why is it (i assume) passing there but not here?

@mixmix

Reply:
on call with mix. it probably worked wrong/test shouldn't be passing in tribes1. alternatives here:
1. query for the latest msg
2. there's a function maybe that calls back every time there's a new msg. `ssb.db.onMsgAdded` You don't even need the cb, you can get it with .value (it's an auto updating observable)